h/static/scripts/components/AnnotationCard.tsx

```diff
@@ -30,14 +30,38 @@ import { callAPI } from '../util/api';
 import { APIError } from '../util/api';
 import AnnotationDocument from './AnnotationDocument';
 
+export type StatusChangeResult =
+  | {
+      saveState: 'saving';
+    }
+  | {
+      saveState: 'saved';
+      newModerationStatus: ModerationStatus;
+    }
+  | {
+      saveState: 'error';
+      error: unknown;
```

Comment:
Is `error` an `Error` object here? We usually write types assuming that promises will always be rejected with `Error`s, even though the language does not require that.

Reply:
Yeah, I would assume it should always be an Error object, but I will have to do a casting, as according to TS, the captured error is `unknown`.